challenges.yml

```diff
@@ -875,3 +875,153 @@ extending_v_for_key:
           as it's not duplicated. Vue will work its magic under the hood so that
           items get rendered correctly
       correct: b
+
+extending_prop_trap:
+  question: |
+    Imagine you have this component that renders the name of a Pizza object,
+    which you can in turn modify by typing text into an input element:
+
+    ```vue
+    <template>
+        <label>Name:</table>
+        <pizza-name-input :pizza="pizza" />
+    </template>
+
+    <script>
+    import { getPizza } from '@/services/get-pizza';
+    import PizzaNameInput from '@/components/pizza-name-input';
+
+    export default {
+        name: 'PizzaFormField',
+        components: {
+            PizzaNameInput,
+        },
+        data() {
+            pizza: null,
+        },
+        async created() {
+            this.pizza = (await getPizza()).data['hydra:member'];
+        },
+    };
+    </script>
+    ```
+
+    ```vue
+    <template>
+        <input type="text" v-model="pizza.name" />
+    </template>
+
+    <script>
+    export default {
+        name: 'PizzaNameInput',
+        props: {
+            pizza: {
+                type: Object,
+                required: true,
+            },
+        },
+    };
+    </script>
+    ```
+
+    What is the mistake we've just made here?
+  explanation: |
+    Our code will *still* work in this situation. But if ever we decide to modify
+    the object in the parent component, our changes will be wiped out and be lost.
```

Comment:
Is this true? We're modifying the `pizza` object by reference... so I think that it *would* actually modify the `pizza` object in the parent component. It's just not "how things are supposed to work" so you're playing with fire by taking this shortcut.

Reply:
I think you are right if we change individual components. But if you change the entire object, everything down the children will change. I will rephrase this because in reality, changing the entire pizza object would produce the desired effect in this case.